src/graphics.rs

```diff
@@ -1815,7 +2078,8 @@ impl Asset for Texture {
                 let mut filesystem = resources.fetch_mut::<Filesystem>();
                 let mut gfx = resources.fetch_mut::<Graphics>();
                 let mut file = filesystem.open(path)?;
-                let texture = Texture::from_reader(&mut *gfx, &mut file)?;
+                let texture = Texture::from_reader(&mut *gfx, &mut file)
+                    .with_context(|| format!("Failed to create a texture using {:?}", path))?;
```

Comment:
prefer `anyhow!` to `format!` here. `String` will happily get converted into an `Error` but it doesn't make me happy, personally.